website/content/docs/discovery/checks.mdx

```diff
@@ -440,70 +535,135 @@ check = {
+   the [initial health check status](#initial-health-check-status) section.
+  
+  -> **Health defaults to critical:** If health status it not initially specified,
+     it defaults to "critical" to protect against including a service
+     in discovery results before it is ready. 
+
+- `deregister_critical_service_after` `(string: "")` - If specified,
+  the associated service and all its checks are deregistered
+  after this check is in the critical state for more than the specified value.
+  The value has the same formatting specification as the [`interval`](#interval) field.
+
+  The minimum timeout is 1 minute,
+  and the process that reaps critical services runs every 30 seconds,
+  so it may take slightly longer than the configured timeout to trigger the deregistration.
+  This field should generally be configured with a timeout that's significantly longer than
+  any expected recoverable outage for the given service.
+
+- `notes` `(string: "")` - Provides a human-readable description of the check.
+  This field is opaque to Consul and can be used however is useful to the user.
+  For example, it could be used to describe the current state of the check.
+
+- `token` `(string: "")` - Specifies an ACL token used for any interaction
+  with the catalog for the check, including
+  [anti-entropy syncs](/docs/architecture/anti-entropy) and deregistration.
+
+  For alias checks, this token is used if a remote blocking query is necessary to watch the state of the aliased node or service.
+
+#### Success/Failures before passing/warning/critical
+
+To prevent flapping health checks and limit the load they cause on the cluster,
+a health check may be configured to become passing/warning/critical only after a
+specified number of consecutive checks return as passing/critical.
+The status does not transition states until the configured threshold is reached.
 
-- Exit code 0 - Check is passing
-- Exit code 1 - Check is warning
-- Any other code - Check is failing
+- `success_before_passing` - Number of consecutive successful results required
+  before check status transitions to passing. Defaults to `0`. Added in Consul 1.7.0.
 
-This is the only convention that Consul depends on. Any output of the script
-will be captured and stored in the `output` field.
+- `failures_before_warning` - Number of consecutive unsuccessful results required
+  before check status transitions to warning. Defaults to the same value as that of
+  `failures_before_critical` to maintain the expected behavior of not changing the
+  status of service checks to `warning` before `critical` unless configured to do so.
+  Values higher than `failures_before_critical` are invalid. Added in Consul 1.11.0.
+
+- `failures_before_critical` - Number of consecutive unsuccessful results required
+  before check status transitions to critical. Defaults to `0`. Added in Consul 1.7.0.
+
+This feature is available for HTTP, TCP, gRPC, Docker & Monitor checks.
```

Comment:
@jkirschner-hashicorp : check with engineering about what a "Monitor" check is...

Reply:
This refers to script type checks